### DISCUSSION TITLE
PortsViewInterface Upgrade

### DISCUSSION THREAD

**Discussion by dewiwynwilliams** [clientIO/joint#2392] [Q&A]
### Introduction

I'm trying to upgrade an application from an old jointjs version (0.9.4) to version 3.7.5. 
The application uses PortsViewInterface which was deprecated in version 3.  
Are there any upgrade guidelines on how to replace PortsViewInterface?

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS

**Comment by kumilingus** [comment]
You don't need the `PortsViewInterface` anymore. Ports can be added to any element from version `v3.0` using the [Ports API](https://resources.jointjs.com/docs/jointjs/v3.7/joint.html#dia.Element.ports).

You might want to switch to [shapes.devs](https://resources.jointjs.com/docs/jointjs/v3.7/joint.html#shapes.devs) which were created to ease the transition (they have `inPorts` and `outPorts` predefined).
